File: backend/ecommerce-service/src/main/java/com/travelplan/ecommerce/entity/Order.java
package com.travelplan.ecommerce.entity;

import com.travelplan.ecommerce.enums.DeliveryType;
import com.travelplan.ecommerce.enums.OrderStatus;
import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tourist_id", nullable = false)
    private String touristId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private OrderStatus status = OrderStatus.PENDING;

    @Column(name = "total_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal totalAmount = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Column(name = "delivery_type", length = 50)
    private DeliveryType deliveryType = DeliveryType.LOCAL_COLLECTION;

    @Column(name = "delivery_location_id")
    private String deliveryLocationId;

    @Column(name = "itinerary_id")
    private String itineraryId;

    // Link to the multiple items in the cart
    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items = new ArrayList<>();

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/order.ts
export type OrderStatus = 'PENDING' | 'CONFIRMED' | 'PROCESSING' | 'SHIPPED' | 'DELIVERED' | 'CANCELLED'
export type DeliveryType = 'PICKUP' | 'DELIVERY' | 'DIGITAL'

export interface OrderItem {
  id: number
  productId: number
  productName: string
  quantity: number
  unitPrice: number
  totalPrice: number
}

export interface Order {
  id: number
  touristId: string
  status: OrderStatus
  deliveryType: DeliveryType
  deliveryAddress?: string
  totalAmount: number
  items: OrderItem[]
  createdAt: string
  updatedAt: string
}

export interface CreateOrderRequest {
  deliveryType: DeliveryType
  deliveryAddress?: string
  items: { productId: number; quantity: number }[]
}
